### ISSUE TITLE
Add support for custom format validators

### ISSUE THREAD

**Issue by gandrisl** [RSuter/NJsonSchema#480]
It would be nice if there would be a way to implement validators to support custom formats.

**Comment by RSuter** [comment]
Good idea, but Im not sure if this is required: Just implement an own class inheriting from to JsonSchemaValidator and override protected Validate(...). There you can validate custom formats or call the base impl

**Comment by gandrisl** [comment]
I think it would be much easier to have some interface for validators (i.e. IJsonSchemaFormatValidator) and have possibility to add them to the validator instance. If that's too much, then there could be a protected virtual method `ValidateFormat` in `JsonSchemaValidator` that would contain format validations now existing in ValidateString method.
